### PR TITLE
Add version createdon metadata 453

### PR DESCRIPTION
**Pull request by carlesarnal**

### CODE REVIEW

File: app/src/main/java/io/apicurio/registry/storage/impl/AbstractMapRegistryStorage.java [hunk 1]
```diff
@@ -199,8 +200,6 @@ protected ArtifactMetaDataDto createOrUpdateArtifact(String artifactId, Artifact
         contents.put(MetaDataKeys.CREATED_ON, currentTimeMillis);
         contents.put(MetaDataKeys.MODIFIED_ON, currentTimeMillis);
```
Comment (EricWittmann) [inline app/src/main/java/io/apicurio/registry/storage/impl/AbstractMapRegistryStorage.java:201]: I don't think this is useful meta-data to put on a version.  The **artifact** has a logical `modifiedOn` ... but similar to the `createdOn` value for an artifact, the `modifiedOn` for an artifact should be the `createdOn` value of the artifact's most recent version.

Comment (carlesarnal) [reply app/src/main/java/io/apicurio/registry/storage/impl/AbstractMapRegistryStorage.java:201]: I can fix it, but nothing has changed regarding the `modifiedOn` metadata here.

File: app/src/main/java/io/apicurio/registry/rest/ArtifactsResourceImpl.java [hunk 2]
```diff
@@ -451,6 +453,12 @@ public void deleteArtifactRule(RuleType rule, String artifactId) {
     @Override
     public ArtifactMetaData getArtifactMetaData(String artifactId) {
         ArtifactMetaDataDto dto = storage.getArtifactMetaData(artifactId);
+
+        if (dto.getVersion() != ARTIFACT_FIRST_VERSION) {
+            //Fetch the createdOn value from the first version of the artifact
+            ArtifactVersionMetaDataDto artifactVersionMetaData = storage.getArtifactVersionMetaData(artifactId, ARTIFACT_FIRST_VERSION);
+            dto.setCreatedOn(artifactVersionMetaData.getCreatedOn());
+        }
```
Comment (EricWittmann) [inline app/src/main/java/io/apicurio/registry/rest/ArtifactsResourceImpl.java:461]: Doesn't this duplicate the logic in the storage?  It seems like the storage is doing this same work (and I think the storage is where we would want to do this work).

Comment (carlesarnal) [reply app/src/main/java/io/apicurio/registry/rest/ArtifactsResourceImpl.java:461]: Yes, this is part of a previous approach and is something that needs to be removed sorry for that.

**Pull request merged**